### PR TITLE
Support for boolean types #1645

### PR DESCRIPTION
**Pull request by kshitij10496**
@stefanv I have tried to fix the issue using `astype`

### CODE REVIEW

File: skimage/segmentation/boundaries.py [hunk 1]
```diff
@@ -199,6 +212,8 @@ def mark_boundaries(image, label_img, color=(1, 1, 0),
     --------
     find_boundaries
     """
+    if image.dtype == 'bool':
+        image = image.astype(np.uint8)
```
Comment (jni) [inline skimage/segmentation/boundaries.py]: This isn't needed. `find_boundaries` will take care of bool images where necessary.

**Comment by jni** [comment]
@kshitij10496 nice, clean work! You should remove a bool -> uint8 conversion in `mark_boundaries` that's not needed, but otherwise I think this is ready! (The travis failures seem to be related to docs building, rather than your changes...)

File: skimage/segmentation/boundaries.py (hunk 1 above)
Comment (kshitij10496) [inline skimage/segmentation/boundaries.py]: @jni Thanks.
I am working on it

**Comment by kshitij10496** [comment]
@jni Removed the bool conversion in `mark_boundaries`

**Comment by jni** [comment]
@kshitij10496 Good work! As I mentioned, the travis failures are unrelated, and I've verified that your tests pass on all versions, so we are good to go! Thank you for your contribution!

**Comment by kshitij10496** [comment]
@jni I am intrigued by image processing and computer vision.
And I am really interested in contributing more to this project.

**Pull request merged**